src/javascript/core.js

```diff
@@ -83,35 +45,48 @@ function track(config, callback) {
 	if (utils.isUndefined(callback)) {
 		callback = function () {};
 	}
-
-	const coreContext = settings.get('config') && settings.get('config').context || {};
```

Comment:
I had troubles tracing out the various directional merges happening across settings context/config context/default config at the top/parameters.  I hope the new layout makes order of overrides clearer.